fs-storage/src/file_storage.rs

```diff
@@ -176,12 +149,100 @@ where
                 ),
             ));
         }
+
+        Ok(data)
+    }
+}
+
+impl<K, V> BaseStorage<K, V> for FileStorage<K, V>
+where
+    K: Ord
+        + Clone
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + std::str::FromStr,
+    V: Clone
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + std::str::FromStr
+        + Monoid<V>,
+{
+    /// Set a key-value pair in the internal mapping
+    fn set(&mut self, key: K, value: V) {
+        self.data.entries.insert(key, value);
+        self.modified = std::time::SystemTime::now();
+    }
+
+    /// Remove an entry from the internal mapping given a key
+    fn remove(&mut self, id: &K) -> Result<()> {
+        self.data.entries.remove(id).ok_or_else(|| {
+            ArklibError::Storage(self.label.clone(), "Key not found".to_owned())
+        })?;
+        self.modified = std::time::SystemTime::now();
+        Ok(())
+    }
+
+    /// Compare the timestamp of the storage file
+    /// with the timestamp of the in-memory storage and the last written
+    /// to time to determine if either of the two requires syncing.
+    fn sync_status(&self) -> Result<SyncStatus> {
+        let file_updated = fs::metadata(&self.path)?.modified()?;
+
+        // mapping updated
+        let status = if self.modified > self.written_to_disk {
+            // file updated since last write
+            if file_updated > self.written_to_disk {
+                // both updated
```

Comment:
We are only here if the user manually modifies the storage file on disk, correct?

Reply:
Externally modified yes. Either by the user or suppose by a mirror storage (like in the test).